src/ConfigManager.cpp

```diff
@@ -190,5 +184,84 @@ std::string ConfigManager::validateConfig(std::string configName)
 	}
 	// Write an updated config file
 	updatedFile.write(updatedIni);
-	return updatedConfigName;
-}
\ No newline at end of file
+	return updatedUserConfigPath;
+}
+
+void ConfigManager::update()
+{
+	// Clear the sets
+	differentFilePaths.clear();
+	identicalFilePaths.clear();
+	defaultFilenames.clear();
+
+	// Get the filenames of the original config
+	DIR*		   dir;
+	struct dirent* ent;
+	if ((dir = opendir(originalConfigDir.c_str())) != NULL)
+	{
+		while ((ent = readdir(dir)) != NULL)
+		{
+			std::string filename = ent->d_name;
+			if (filename != "." && filename != "..")
+			{
+				defaultFilenames.insert(filename);
+			}
+		}
+		closedir(dir);
+	}
+	else
+	{
+		exit(18886);
+	}
+
+	for (auto& filename : defaultFilenames)
+	{
+		std::string originalFile = originalConfigDir + "/" + filename;
+		std::string userFile	 = userConfigDir + "/" + filename;
+
+		if (compareFiles(originalFile, userFile))
+		{
+			identicalFilePaths.insert(filename);
+		}
+		else
+		{
+			differentFilePaths.insert(filename);
+		}
+	}
+}
+
+void ConfigManager::computeFullPathAndName()
+{
+	if ((userConfigPath.rfind("./", 0) == 0) || (userConfigPath.rfind("/", 0) == 0))
+	{
+		if ((userConfigPath.rfind("./", 0) == 0))
+		{
+			char buffer[PATH_MAX];
+#ifdef _WIN32
+			if (GetCurrentDirectory(PATH_MAX, buffer))
+			{
+				return (buffer + userFilePath.substr(1));
+			}
+			else
+			{
+				throw std::runtime_error("GetCurrentDirectory() error: " +
+										 std::to_string(GetLastError()));
+			}
+#else
+			if (getcwd(buffer, sizeof(buffer)) != NULL)
+			{
+				userConfigPath = (buffer + userConfigPath.substr(1));
+			}
+			else
+			{
+				perror("getcwd() error");
+			}
+#endif
+		}
+	}
+	else
+	{
+		userConfigPath = (userConfigDir + "/" + userConfigPath);
+	}
+	userConfigName = userConfigPath.substr(userConfigPath.find_last_of("/") + 1);
```

Comment:
when loading config files (from path), only open files with .cfg extension! 

Btw consider check of filesize, if larger than 1M throw error that config seems wrong. Ive accidentally selected sample.txt (1GB random value file) and it added default fields to it and tired to upload it to md80 xd